pkg/resource/table/hooks_replica_updates.go

```diff
@@ -0,0 +1,406 @@
+	defer func() {
+		exit(err)
+	}()
+
+	createReplicas, updateReplicas, deleteRegions := calculateReplicaUpdates(latest, desired)
+
+	input = &svcsdk.UpdateTableInput{
+		TableName:      aws.String(*desired.ko.Spec.TableName),
+		ReplicaUpdates: []svcsdktypes.ReplicationGroupUpdate{},
+	}
+
+	totalReplicasOperations := len(createReplicas) + len(updateReplicas) + len(deleteRegions)
+	replicasInQueue = totalReplicasOperations - 1
+
+	// Process replica updates in order: create, update, delete
+	// We'll only perform one replica action at a time
+
+	if len(createReplicas) > 0 {
+		region := *createReplicas[0].RegionName
+		rlog.Debug("creating replica in region", "table", *desired.ko.Spec.TableName, "region", region)
+		input.ReplicaUpdates = append(input.ReplicaUpdates, createReplicaUpdate(createReplicas[0]))
+		return input, replicasInQueue, nil
+	}
+
+	if len(updateReplicas) > 0 {
+		region := *updateReplicas[0].RegionName
+		rlog.Debug("updating replica in region", "table", *desired.ko.Spec.TableName, "region", region)
+		input.ReplicaUpdates = append(input.ReplicaUpdates, updateReplicaUpdate(updateReplicas[0]))
+		return input, replicasInQueue, nil
+	}
+
+	if len(deleteRegions) > 0 {
+		region := deleteRegions[0]
+		rlog.Debug("deleting replica in region", "table", *desired.ko.Spec.TableName, "region", region)
+		input.ReplicaUpdates = append(input.ReplicaUpdates, deleteReplicaUpdate(deleteRegions[0]))
+		return input, replicasInQueue, nil
+	}
+
+	return input, replicasInQueue, nil
+}
+
+// calculateReplicaUpdates calculates the replica updates needed to reconcile the latest state with the desired state
+// Returns three slices: replicas to create, replicas to update, and region names to delete
+func calculateReplicaUpdates(
+	latest *resource,
+	desired *resource,
+) (
+	createReplicas []*v1alpha1.CreateReplicationGroupMemberAction,
+	updateReplicas []*v1alpha1.CreateReplicationGroupMemberAction,
+	deleteRegions []string,
+) {
+	existingRegions := make(map[string]*v1alpha1.CreateReplicationGroupMemberAction)
+	if latest != nil && latest.ko.Spec.TableReplicas != nil {
+		for _, replica := range latest.ko.Spec.TableReplicas {
+			if replica.RegionName != nil {
+				existingRegions[*replica.RegionName] = replica
+			}
+		}
+	}
+
+	desiredRegions := make(map[string]*v1alpha1.CreateReplicationGroupMemberAction)
+	if desired != nil && desired.ko.Spec.TableReplicas != nil {
+		for _, replica := range desired.ko.Spec.TableReplicas {
+			if replica.RegionName != nil {
+				desiredRegions[*replica.RegionName] = replica
+			}
+		}
+	}
+
+	// Calculate replicas to create or update
+	for regionName, desiredReplica := range desiredRegions {
+		existingReplica, exists := existingRegions[regionName]
+		if !exists {
+			createReplicas = append(createReplicas, desiredReplica)
+		} else if !equalCreateReplicationGroupMemberActions(existingReplica, desiredReplica) {
+			updateReplicas = append(updateReplicas, desiredReplica)
+		}
+	}
```

Comment:
nit: we can improve variable naming here:
```suggestion
	for desiredRegion, desiredReplica := range desiredRegions {
		existingReplica, exists := existingRegions[desiredRegion]
		if !exists {
			createReplicas = append(createReplicas, desiredReplica)
		} else if !equalCreateReplicationGroupMemberActions(existingReplica, desiredReplica) {
			updateReplicas = append(updateReplicas, desiredReplica)
		}
	}
```